src/oaks_speech.c

```diff
@@ -0,0 +1,2237 @@
+}
+
+static void OakSpeech_StopHighlightedGenderFrameBlink(OakSpeechData *data) {
+    u16 plttBuf[2];
+
+    plttBuf[0] = data->genderSelectFrameDefaultPalette;
+    plttBuf[1] = RGB(27, 28, 28);
+    BG_LoadPlttData(GF_BG_LYR_SUB_2, plttBuf, sizeof(plttBuf), sButtonBlinkPalOffsets[0] * 2);
+    BG_LoadPlttData(GF_BG_LYR_SUB_2, plttBuf, sizeof(plttBuf), sButtonBlinkPalOffsets[1] * 2);
+}
+
+static BOOL OakSpeech_GenderSelectHandleInput(OakSpeechData *data) {
+    BOOL ret = FALSE;
+    int hitbox;
+
+    TouchscreenHitbox hitboxes[3];
+    ARRAY_ASSIGN(hitboxes, sTouchscreenHitboxes_GenderSelect);
+
+    if (gSystem.touchNew) {
+        hitbox = TouchscreenHitbox_FindRectAtTouchNew(hitboxes);
+        if (hitbox != -1) {
+            data->menuData.cursorPos      = hitbox;
+            data->menuData.pressDelay     = 1;
+            data->menuData.flashFramesPer = 2;
+            OakSpeech_BlinkHighlightedGenderFrame(data, 1);
+            PlaySE(SEQ_SE_DP_SELECT);
+            data->lastChosenGender = hitbox;
+            ret                    = TRUE;
+        }
+    } else if (!data->menuData.inPadMode) {
+        if (gSystem.newKeys & (PAD_BUTTON_A | PAD_KEY_LEFT | PAD_KEY_RIGHT)) {
+            PlaySE(SEQ_SE_DP_SELECT);
+            data->menuData.inPadMode = 1;
+            OakSpeech_BlinkHighlightedGenderFrame(data, 1);
+        }
+    } else {
+        OakSpeech_BlinkHighlightedGenderFrame(data, 0);
+        if (gSystem.newKeys & PAD_KEY_LEFT) {
+            if (data->menuData.cursorPos != 0) {
+                --data->menuData.cursorPos;
+                PlaySE(SEQ_SE_DP_SELECT);
+            }
+        } else if (gSystem.newKeys & PAD_KEY_RIGHT) {
+            if (data->menuData.cursorPos != data->menuData.numOptions - 1) {
+                ++data->menuData.cursorPos;
+                PlaySE(SEQ_SE_DP_SELECT);
+            }
+        } else if (gSystem.newKeys & PAD_BUTTON_A) {
+            data->menuData.inPadMode      = 0;
+            data->menuData.pressDelay     = 1;
+            data->menuData.flashFramesPer = 2;
+            OakSpeech_BlinkHighlightedGenderFrame(data, 1);
+            PlaySE(SEQ_SE_DP_SELECT);
+            ret                    = TRUE;
+            data->lastChosenGender = data->menuData.cursorPos;
+        }
+    }
+
+    return ret;
+}
+
+static void OakSpeech_InitPlayerPicShrinkAnim(OakSpeechData *data) {
+    data->playerPicShrinkAnimStep  = 0;
+    data->playerPicShrinkAnimDelay = 0;
+}
+
+static BOOL OakSpeech_PlayerPicShrinkAnim(OakSpeechData *data) {
+    BOOL ret = FALSE;
+    int gfxId;
+
+    if (data->playerPicShrinkAnimDelay != 0) {
+        --data->playerPicShrinkAnimDelay;
+    } else {
+        ++data->playerPicShrinkAnimStep;
+        data->playerPicShrinkAnimDelay = 8;
+    }
+    if (data->playerGender == PLAYER_GENDER_MALE) {
+        int sp28[6];
+        ARRAY_ASSIGN(sp28, sPlayerPicShrinkGfx_Male);
+        gfxId = sp28[data->playerPicShrinkAnimStep];
+    } else {
+        int sp10[6];
```

Comment:
name sp10

Reply:
this is a fakematch so the name can stay